net/lp_rpc.proto

```diff
@@ -69,9 +122,38 @@ message SegData {
 // Individual transcoded segment data.
 message TranscodedSegmentData {
 
-    // URL where the transcoded data can be downloaded from
+    // URL where the transcoded data can be downloaded from.
     string url = 1;
 
+    TypedURI turl = 2;
+}
+
+/*
+  TypedURI contains information needed to access segment written to specific storage.
+  Plus it contains presentation of this segment in the manifest.
+  Potentially it can contain credentials for reading data from storage
+  (for now all we write should be public).
+  */
+message TypedURI {
```

Comment:
It is attempt to give explicit structure to data contained in urls. Without that we start writing functions like `parseStreamID`, which attempts to extract some data contained in url. 
`storage` allows us to do things like `drivers.IsOwnStorage(typedURI)`, `drivers.GetSegmentData(typedURI` etc.
`uri_in_manifest` - actually used - `mseg.URI = turi.UriInManifest` - it does what it says, contains value that should be put into URI field of manifest. It can be just segment name, it can be absolute http url. Initially for IPFS it was `ipfs://hash`, but now we want store data in ipfs folder, so for IPFS it will be just file name.
`streamID` - needed to put segment's data into right place (or else we should parse this from segment's name)
`seqNo` - that one can be removed, but then we need to add it as argument to `InsertLink` - because we need that data when we're inserting new entry into manifest.

Reply:
> uri_in_manifest - actually used - mseg.URI = turi.UriInManifest

Still not clear why this is needed over the plain URL field. When would the broadcaster need a hint from the transcoder what goes into the manifest that's different from the other URL field?

> Initially for IPFS it was ipfs://hash, but now we want store data in ipfs folder

Maybe I'm misunderstanding, but don't we need both?

> drivers.IsOwnStorage(typedURI)

This is only needed for checking S3 right? We just need the URI for that. The current implementation is quite close, but something like:
```golang
func IsOwnStorage(uri *URL) {
  if s3Info != nil && uri.Host.HasPrefix(s3Info.Bucket+".s3-"+s3Info.Region) return true
  return false
}
```

> drivers.GetSegmentData(typedURI

The only time this would be different than a regular download would be for IPFS, right? Would something like this work?

```golang 
func GetSegmentData(uri *URL) {
  if uri.Scheme == "ipfs" {
    // download ipfs
  }
 // download http
}

I can see us needing some more structure around the wire format at some point, but I don't see it right now. Best to avoid imposing excessive structure up-front when we can safely achieve our desired behavior using the existing format.

> seqNo - that one can be removed, but then we need to add it as argument to InsertLink

Both the sender and receiver should already have the sequence number information.